docs/03-03-examples-alpha.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Kyma CLI alpha command usage examples
+type: Details
+---
+
+The following examples show how to provision a cluster, deploy Kyma, and <!--- update is not among the alpha commands ---> with alpha commands.
+
+## Provision a cluster locally or using cloud providers
+<!---provision gke is not among the alpha commands--->
+To provision a cluster on a specific cloud provider (in this example GCP), run:
+
+```bash
+kyma alpha provision gke --credentials {SERVICE_ACCOUNT_KEY_FILE_PATH} --name {CLUSTER_NAME} --project {GCP_PROJECT} 
+```
+
+To provision a Minikube cluster, run:
+<!---provision minikube is not among the alpha commands--->
+```bash
+kyma alpha provision minikube
+```
+
+## Deploy Kyma
+
+To deploy Kyma using your own domain, run:
+
+```bash
+kyma alpha deploy --domain {DOMAIN} --tls-cert {TLS_CERT} --tls-key {TLS_KEY}
+```
+
+`{TLS_CERT}` and `{TLS_KEY}` are the TLS certificate and TLS key for the domain used. Note that both of them must be encoded in base64. You can use [OpenSSL](https://www.openssl.org/) (which is installed by default with macOS) to convert your TLS certificate and TLS key to base64-encoded values.
+
+You can simplify the process to deploy Kyma with your own domainin  `{DOMAIN}` the following way:
+
+1. Encode the TLS certificate and TLS key, and export them as environment variables:
+  
+    ```bash
+    export TLS_CERT="$(openssl base64 -in {TLS_CERT_FILE_PATH})"
+    export TLS_KEY="$(openssl base64 -in {TLS_KEY_FILE_PATH})"
+    ```
+
+    `{TLS_CERT_FILE_PATH}` is the path to the file containing the TLS certificate, and `{TLS_KEY_FILE_PATH}` is the path to the file containing the TLS key.
+
+2. Use these environment variables in the command:
+
+    ```bash
+    kyma alpha deploy --domain {DOMAIN} --tls-cert $TLS_CERT --tls-key $TLS_KEY
+    ```
+
+To deploy Kyma from the `master` branch, run:
+
+```bash
+kyma alpha deploy --source master
+```
+
+To deploy Kyma using your own Kyma installer image, run:
+
+```bash
+kyma alpha deploy --source {IMAGE}
+```
+
+To build an image from your local sources and deploy Kyma on a remote cluster based on this image, run:
+
+```bash
+kyma alpha deploy --source local --custom-image {IMAGE}
+```
+>    **EXAMPLE:** If your `{IMAGE}` is `user/my-kyma-installer:v1.4.0`, this command builds an image from your local sources and push it your repository `user/my-kyma-installer` with the tag `v1.4.0`.<br>
+Then, this image `user/my-kyma-installer:v1.4.0` is used in the `Deployment` for `kyma-installer`.
+
+To deploy kyma with only specific components, run:
+
+```bash
+kyma alpha deploy --components {COMPONENTS_FILE_PATH}
+```
+`{COMPONENTS_FILE_PATH}` is the path to a YAML file containing the desired component list to be installed, for example:
+```bash
+  components:
+  - name: "cluster-essentials"
+    namespace: "kyma-system"
```

Comment:
Please remove the neamesace entries having "kyma-system" as this namespace will be used by default, no need to specify it

Reply:
Here an example of the latest components file: https://github.com/kyma-project/kyma/blob/master/installation/resources/components.yaml

Maybe we need a short explanation how the structure of this file has to be (@a-thaler : maybe worth to think about a schema validation in the CLI which detects invalid formats).

Reply:
What I got now is:

o deploy Kyma with only specific components, run:

```bash
kyma alpha deploy --components {COMPONENTS_FILE_PATH}
```
`{COMPONENTS_FILE_PATH}` is the path to a YAML file containing the desired component list to be installed, for example:
```bash
  components:
  - name: "cluster-essentials"
  - name: "testing"
  - name: "istio"
    namespace: "istio-system"
  - name: "xip-patch"
    namespace: "kyma-installer"
  - name: "istio-kyma-patch"
    namespace: "istio-system"
  - name: "dex"
  ```
  
I'll need input for the explanation that Tobias suggested.